pkg/tempopb/prealloc.go

```diff
@@ -33,7 +33,7 @@ func (r *PreallocBytes) Unmarshal(dAtA []byte) error {
 // MarshalTo implements proto.Marshaller.
 // returned int is not used
 func (r *PreallocBytes) MarshalTo(dAtA []byte) (int, error) {
-	copy(dAtA[:], r.Slice[:])
+	copy(dAtA, r.Slice)
```

Comment:
it's really strange we were reslicing the entire slice before? i'm wondering if this was a holdover from a previous array -> slice conversion